### DISCUSSION TITLE
timeoutSeconds not working for retries

### DISCUSSION THREAD

**Discussion by dapeng09** [Netflix/conductor#3830] [Q&A]
Hi, I am new to Conductor. 
If I understand right, when a worker task (attempt0) timeouts and gets retried according to my task definition, the timeoutSeconds should apply as well to the attempt1 and attempt2... but it looks to me it's not working this way. What am I missing here? Thanks.

My worker is sleeping for 10 minutes, attempt0 timeouts as expected, but attempt1 stays in IN_PROGRESS for 10 minutes and finally gets completed.

**workflow definition**
```
{
    "name": "timeout_test_task",
    "description": "timeout test task",
    "version": 1,
    "tasks":
    [
        {
            "name": "step_1_task",
            "taskReferenceName": "step_1_task_ref",
            "type": "SIMPLE"
        }
    ],
    "restartable": true,
    "ownerEmail": "abc@example.com",
    "workflowStatusListenerEnabled": true,
    "schemaVersion": 2
}
```
**task definition**
```
    {
        "name": "step_1_task",
        "retryCount": 5,
        "timeoutSeconds": 10,
        "pollTimeoutSeconds": 1200,
        "timeoutPolicy": "RETRY", 
        "retryLogic": "FIXED",
        "retryDelaySeconds": 1,
        "responseTimeoutSeconds": 5,
        "concurrentExecLimit": 100,
        "rateLimitFrequencyInSeconds": 60,
        "ownerEmail": "abc@example.com",
        "rateLimitPerFrequency": 2
    }
```
**UI screenshot of attempt0**
![image](https://github.com/Netflix/conductor/assets/4626700/c55e96da-7e88-4398-8389-503dee555cdb)

**UI screenshot of attempt1**
![image](https://github.com/Netflix/conductor/assets/4626700/bceb9379-d428-40b1-b1c0-1cdc2e1fedf5)

BTW, attempt0 timeouts after 31 seconds instead of 10 seconds, what's causing this delay?

**Comment by manan164** [comment]
Hi @dapeng09 , The workflow will get evaluated after every 30 seconds. If you want lesser granularity then please set the environment property `conductor.app.workflowOffsetTimeout` to lesser value.

**Reply by dapeng09** [reply]
Hi @manan164, thanks for the help, I will take a shot. Any ideas about timeoutSeconds not taking effect for retry attempt1? Is that how it's designed?

**Reply by manan164** [reply]
Hi @dapeng09 , Let me have a look.

**Reply by dapeng09** [reply]
hi @manan164 Any clue on this?
